.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+version: 2.1
+
+orbs:
+  aws-eks: circleci/aws-eks@1
+  gcp-cli: circleci/gcp-cli@1
+  go: circleci/go@1
+
+jobs:
+  checkout:
+    executor:
+      name: go/default
+      tag: '1.17'
+    steps:
+      - checkout
+      - go/mod-download-cached
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - go
+            - project
+
+  build:
+    executor:
+      name: go/default
+      tag: '1.17'
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Building package
+          command: make build
```

Comment:
Would we want to run codegen and build? If so, we can get away with just `make`

Reply:
Running codegen would be useful, I agree, but only if we act on the output. E.g. fail the build if this step generates a diff, meaning devs need to commit the output of codegen.

Reply:
This would be doable. `_output` is in the `.gitignore` so it shouldn't create any false positives.

Reply:
IMO `codegen` should not be run in the CI, especially at the `build` stage. Maybe it could be run in the `test` stage where the build fails if a diff i generated as @antoineco  mentioned.

Reply:
I am merging the PR because of the tight schedule.